src/index.test.ts

```diff
@@ -2012,7 +1972,19 @@ describe('decodeMappingKey', () => {
       ],
     );
     assert.deepStrictEqual(
-      erc725Instance.decodeMappingKey(
+      decodeMappingKey(
+        encodeKeyName('MyKeyName:<bytes16>', ['0x12345678']),
+        'MyKeyName:<bytes16>',
+      ),
+      [
+        {
+          type: 'bytes16',
+          value: '00000000000000000000000012345678',
```

Comment:
This is incorrect, `bytesN` are right padded, so it should be:

```suggestion
          value: '12345678000000000000000000000000',
```

Reply:
I tried to fix various things, but it really went down the rabbit hole. I would like to talk on Monday. I created a new PR which shows where I found/fixed problems.